src/isomorphic/classic/element/ReactElement.js

```diff
@@ -52,17 +57,17 @@ if (__DEV__) {
  */
 var ReactElement = function(type, key, ref, self, source, owner, props) {
   var element = {
+    // This tag allow us to uniquely identify this as a React Element
+    $$typeof: TYPE_SYMBOL,
+
     // Built-in properties that belong on the element
     type: type,
     key: key,
     ref: ref,
-    self: self,
-    source: source,
```

Comment:
These weren't intended for production use, and definitely not public. So I'm moving this to an underscore, `__DEV__` and making them non-enumerable. Otherwise jest's equality helpers won't work on elements created in two different source locations.